src/scorer.py

```diff
@@ -4,19 +4,7 @@
 import re
 from typing import Tuple, List, Optional
 from collections import Counter
-
-
-from .objdump import objdump, get_arch
-
-
-@dataclass(init=False, unsafe_hash=True)
-class DiffAsmLine:
-    line: str = field(compare=False)
-    mnemonic: str
-
-    def __init__(self, line: str) -> None:
-        self.line = line
-        self.mnemonic = line.split(None, 1)[0]
+from .objdump import ArchSettings, Line, objdump, get_arch
```

Comment:
```suggestion

from .objdump import ArchSettings, Line, objdump, get_arch
```
nit: we leave an empty line between built-in imports and ones from within this module (and third-party libs in between there if any)